ACPs/77-reinventing-subnets/README.md

```diff
@@ -177,41 +272,60 @@ type ExitValidatorSetTx struct {
 }
 ```
 
-#### SetSubnetValidatorManagerTx
+### Proof of Subnet Validator Set Change
 
-To participate in this new registration flow, Subnets must set the `(blockchainID, address)` pair by using `SetSubnetValidatorManagerTx`. This pair will be used to validate the `AddressedCall` in `SetSubnetValidatorWeightTx` and `RegisterSubnetValidatorTx`.
+To track whether a Subnet Validator addition/modification/removal occured on the P-Chain, Subnet Validators must be willing to sign an `AddressedCall` attesting to the validator set change. Since all Subnet Validators sync the P-Chain, only the Subnet Validators need to sign the `AddressedCall`, not all Primary Network Validators.
 
-When a Subnet is created with `CreateSubnetTx`, a validator manager is not specified. To solve this problem, the first `SetSubnetValidatorManagerTx` will expect the Warp `AddressedCall` to have the `sourceChainID` be set to the `SubnetID` and the `sourceAddress` to be zero'd out. Once this transaction is accepted, `RegisterSubnetValidatorTx` and `SetSubnetValidatorWeightTx` are enabled for use on the Subnet.
+Two `AddressedCall`s are defined with the below payloads. For each of them, the `sourceChainID` must be set to the P-Chain ID and the `sourceAddress` must be set to an empty byte array.
 
-```golang
-type SetSubnetValidatorManagerTx struct {
-    // Metadata, inputs and outputs
-    BaseTx
-    // Warp message should include:
-    //   - SubnetID
-    //   - ChainID (where the validator manager lives)
-    //   - Address (address of the validator manager)
-    //   - BLS multisig over the above payload
-    Message warp.Message `json:"message"`
-}
+The method of requesting is left unspecified as it can be more generic. A viable option for supporting this functionality is laid out in [ACP-118](../118-warp-signature-request/README.md) with the `SignatureRequest` message.
+
+#### SubnetValidatorRegistrationMessage
+
+```text
++--------------+----------+----------+
+|      codecID :   uint16 |  2 bytes |
++--------------+----------+----------+
+|       typeID :   uint32 |  4 bytes |
++--------------+----------+----------+
+| validationID : [32]byte | 32 bytes |
++--------------+----------+----------+
+|        valid :     bool |  1 byte  | 
++--------------+----------+----------+
+                          | 39 bytes |
+                          +----------+
 ```
 
-A `typeID` for distinguishing between the different payload types on the P-Chain must be included prior to this ACP being considered "Implementable".
+- `codecID` is the codec version used to serialize the payload and is hardcoded to `0x0000`
+- `typeID` is the payload type identifier and is `0x00000002` for this message
+- `validationID` is the SHA256 of the `Payload` of the `AddressedCall` in the `RegisterSubnetValidatorTx` adding the validator to the Subnet's validator set
+- `valid` indicates whether or not the `validationID` corresponds to a valid `AddressedCall` payload
```

Comment:
IMO the name of the field is confusing as well. Maybe `registered` would make more sense?

Reply:
Yeah it's definitely kinda odd because there are more than two possible states but we're using a `bool`.

The 3 states are:
1. Currently active validation period.
2. Not a currently active validation period, but could possibly become one in the future (not yet passed its registration expiry time)
3. Not a currently active validation period, and will never be an active validation period (due to the expire being passed)

No message should be willing to be signed for validation IDs in case 2, so there isn't a need represent it at all.

I agree `registered` is more clear though.